chat-widget/src/components/notificationpanestateful/NotificationPaneStateful.tsx

```diff
@@ -0,0 +1,112 @@
+import { INotificationPaneInternal } from "@microsoft/omnichannel-chat-components/lib/types/components/notificationpane/interfaces/common/INotificationPaneInternal";
+import { NotificationScenarios } from "../webchatcontainerstateful/webchatcontroller/enums/NotificationScenarios";
+import useChatAdapterStore from "../../hooks/useChatAdapterStore";
+import { ILiveChatWidgetContext } from "../../contexts/common/ILiveChatWidgetContext";
+import { ILiveChatWidgetAction } from "../../contexts/common/ILiveChatWidgetAction";
+import useChatContextStore from "../../hooks/useChatContextStore";
+import { TelemetryHelper } from "../../common/telemetry/TelemetryHelper";
+import { LogLevel, TelemetryEvent } from "../../common/telemetry/TelemetryConstants";
+import { LiveChatWidgetActionType } from "../../contexts/common/LiveChatWidgetActionType";
+import { ConfirmationState } from "../../common/Constants";
+import { defaultChatDisconnectStyleProps } from "./defaultProps/defaultChatDisconnectStyleProps";
+import { defaultChatDisconnectControlProps } from "./defaultProps/defaultChatDisconnectControlProps";
+
+export const NotificationPaneStateful = (props: INotificationPaneStatefulProps) => {
+    const { notificationPaneProps, notificationScenarioType, endChat } = props;
+
+    const [state, dispatch]: [ILiveChatWidgetContext, Dispatch<ILiveChatWidgetAction>] = useChatContextStore();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const [adapter,]: [any, (adapter: any) => void] = useChatAdapterStore();
+    const localConfirmationPaneState = useRef(state?.domainStates?.confirmationState);
+
+    const onCloseChatClick = async () => {
+        TelemetryHelper.logActionEvent(LogLevel.INFO, { Event: TelemetryEvent.NotificationCloseChatButtonClicked, Description: "Notification Close Chat button clicked." });
+
+        if (localConfirmationPaneState.current !== ConfirmationState.Ok) {
+            dispatch({ type: LiveChatWidgetActionType.SET_SHOW_CONFIRMATION, payload: true });
+        } else {
+            const skipEndChatSDK = true;
+            const skipCloseChat = false;
+            const postMessageToOtherTabs = true;
+            await endChat(adapter, skipEndChatSDK, skipCloseChat, postMessageToOtherTabs);
+        }
+    };
+
+    useEffect(() => {
+        localConfirmationPaneState.current = state?.domainStates?.confirmationState;
+    }, [state?.domainStates?.confirmationState]);
+
+    const {useDismissNotification} = hooks;
+    const dismissNotification = useDismissNotification();
+    const handleDismissNotification = useCallback(() => dismissNotification(notificationScenarioType ?? ""), []); 
+
+    let genericPropsObj: INotificationPaneInternal = {};
+
+    // populate INotificationPaneInternal after merging customized props with default props
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const populateInternalProps = (notificationProps: any) => {
```

Comment:
Why is this `any` type?

Reply:
We need to support multiple types of notification interfaces. For example, `notificationProps` needs to be of type `IChatDisconnectNotificationProps`, and in the future `IAttachmentErrorNotificationProps`, `IDownloadTranscriptErrorNotificationProps`, etc. Those different interfaces have slight variations to their properties (`ChatDisconnect` contains `CloseChatButton` related props whereas others do not).